### ISSUE TITLE
scheduler with constrain_start_time results in warning

### ISSUE THREAD

**Issue by snjoetw** [home-assistant/appdaemon#143]
Seeing this in the log:

> 2017-09-20 21:56:38.287416 WARNING ------------------------------------------------------------
> 2017-09-20 21:56:38.287696 WARNING Unexpected error during exec_schedule() for App: back_yard_darkness_lighting_automation_when_evening
> 2017-09-20 21:56:38.287915 WARNING Args: {'name': 'back_yard_darkness_lighting_automation_when_evening', 'id': UUID('013f580a-4508-414a-89e1-5ef8a2446fa4'), 'callback': <bound method DarknessLightingAutomation.entity_state_change of <lighting_automation_darkness.DarknessLightingAutomation object at 0x7fa541fefcc0>>, 'timestamp': 1505969798, 'interval': 60, 'basetime': 1505969798, 'repeat': True, 'offset': 0, 'type': None, 'kwargs': {'interval': 60}}
> 2017-09-20 21:56:38.288060 WARNING ------------------------------------------------------------
> 2017-09-20 21:56:38.288669 WARNING Traceback (most recent call last):
>   File "/usr/local/lib/python3.6/site-packages/appdaemon/appdaemon.py", line 312, in exec_schedule
>     "kwargs": args["kwargs"],
>   File "/usr/local/lib/python3.6/site-packages/appdaemon/appdaemon.py", line 242, in dispatch_worker
>     if not check_time_constraint(conf.app_config[name], name):
>   File "/usr/local/lib/python3.6/site-packages/appdaemon/appdaemon.py", line 228, in check_time_constraint
>     if not utils.now_is_between(start_time, end_time, name):
>   File "/usr/local/lib/python3.6/site-packages/appdaemon/utils.py", line 245, in now_is_between
>     start_time = parse_time(start_time_str, name)
>   File "/usr/local/lib/python3.6/site-packages/appdaemon/utils.py", line 193, in parse_time
>     parts = re.search('^(\d+):(\d+):(\d+)', time_str)
>   File "/usr/local/lib/python3.6/re.py", line 182, in search
>     return _compile(pattern, flags).search(string)
> TypeError: expected string or bytes-like object
> 
> 2017-09-20 21:56:38.288871 WARNING ------------------------------------------------------------
> 2017-09-20 21:56:38.289047 WARNING Scheduler entry has been deleted
> 2017-09-20 21:56:38.289192 WARNING ------------------------------------------------------------
> 

This is what I have in the yaml:

```
back_yard_darkness_lighting_automation_when_evening:
  module: lighting_automation_darkness
  class: DarknessLightingAutomation
  darkness_sensor: sensor.template_backyard_luminance
  darkness_threshold: 20
  entity: switch.backyard_light_switch
  constrain_start_time: 16:00:00
  constrain_end_time: 23:30:00
```

In my app, I have this below which I think is triggering the warning above
```
    self.run_minutely(self.entity_state_change, None)
```

**Comment by snjoetw** [comment]
After looking at the code, it looks like I need to add quotes around the times to make them string. I think that needs to be reflected in the doc as in the example, there's no quotes

**Comment by acockburn** [comment]
Yes, this is likely introduced by the move to YAML which can be a little picky about such things. I'll add a note to the docs.

**Comment by acockburn** [comment]
Fixed the docs in 3.0
